### PR TITLE
Firestore [GCP-7421]: Add docs regarding Client threadsafety

### PR DESCRIPTION
**Pull request by HemangChothani**
Issue [7421]

### CODE REVIEW

File: firestore/docs/index.rst [hunk 1]
```diff
@@ -1,5 +1,12 @@
 .. include:: README.rst
 
+.. note::
+
+   Because the firestore client uses :mod:`grpc` library and the third-party
+   :mod:`requests` library, it is safe to share instances across threads.
+   In multiprocessing scenarios, best practice is to create client instances *after*
+   :class:`multiprocessing.Pool` or :class:`multiprocessing.Process` invokes
+   :func:`os.fork`.
```
Comment (mf2199) [inline firestore/docs/index.rst]: Just a missing article and a bit of rephrasing, to make it a bit clearer:
```
In multiprocessing scenarios, the best practice is to create client instances *after*
the invocation of :func:`os.fork` by :class:`multiprocessing.Pool` or
:class:`multiprocessing.Process`.
```

Comment (HemangChothani) [reply firestore/docs/index.rst]: @mf2199  tres suggest to `see storage/docs/index.rst` for an example which he mentioned in issue's comment so i have copied from there, so please look into it once and if its needed then i will change what  you recommend.

Comment (mf2199) [reply firestore/docs/index.rst]: These docs are made by developers just like ourselves and are never guaranteed to be grammatically perfect. We should, however, do our best effort.

Comment (HemangChothani) [reply firestore/docs/index.rst]: ok ,done.

**Pull request closed without merging**